src/server/api/index.ts

```diff
@@ -94,6 +96,25 @@ async function apiBuilder (cfg: APIConfig): Promise<FastifyTypebox> {
     return 'ok\n'
   })
 
+  // Setup announcements actor
+  const actorUrl = `${cfg.publicURL}/v1/announcements`
+  let keys
+  try {
+    const prev = await store.announcements.getInfo()
+    keys = { ...prev.keypair, publicKeyId: prev.publicKeyId }
+  } catch {
+    keys = { ...generateKeypair(), publicKeyId: `${actorUrl}#${DEFAULT_PUBLIC_KEY_FIELD}` }
+  }
+  await store.announcements.setInfo({
```

Comment:
What's the purpose for re-invoking `setInfo` is it in case the publicURL changes? Might be good to only invoke it inside `catch` or if `!prev.actorURL === actorURL`. Also per the previous comment, mind putting this logic into `apsystem.announcements.init(publicURL)`